QuizApp/App/Screens/Quiz/QuizListViewModel.swift

```diff
@@ -14,16 +15,31 @@ final class QuizListViewModel {
 
 extension QuizListViewModel {
 
-    func fetchQuiz() {
-        Task(priority: .background) {
-            let quizes = try await quizUseCase.fetchQuizes()
+    func fetchQuiz(for category: String) {
+        Task {
+            let quizes = try await quizUseCase.fetchQuizes(for: category)
 
             DispatchQueue.main.async { [weak self] in
                 guard let self = self else { return }
+
                 self.quizes = quizes
                     .map { QuizViewModel(from: $0) }
             }
         }
     }
 
+    func fetchCategories() {
```

Comment:
I don't think it makes sense to fetch all quizzes just so that you could get the categories. I think the preferred solution would be to have sections hardcoded in the app (looping through enum cases.)